### DISCUSSION TITLE
（自動保存はされているが）記事の執筆中にヒストリバックしたときにアラートを出してほしい

### DISCUSSION THREAD

**Discussion by JunichiIto** [increments/qiita-discussions#424] [要望・アイデア・不具合]
タイトルの通りです。記事の執筆中にヒストリバックをしたら、アラートが出てくれると嬉しいです。

<img width="253" alt="Screenshot 2023-01-27 at 11 59 51" src="https://user-images.githubusercontent.com/1148320/215001081-b5068d77-0108-460e-9e87-9d3bc585f95d.png">

## もっと詳しく
昨日、ちょっと以前途中まで書いた記事の続きを書いていたのですが、かなり書き進めたところで、ブラウザの操作を誤って前画面に戻ってしまいました。
「あ、やべっ！」と思って「→」ボタンで元の編集画面に戻ったところ、編集前の状態が表示されて「ひょえ〜〜〜！！！😱」となってしまいました。

再現するとこんな感じです↓

step 1. 下書きの続きを書こうとする
<img width="1034" alt="1" src="https://user-images.githubusercontent.com/1148320/215001737-d28f564f-3d14-4df7-a4bb-0542120dcbe7.png">

step 2. 続きを書く
<img width="1041" alt="2" src="https://user-images.githubusercontent.com/1148320/215001776-209700f9-1c7a-47c5-9b24-f1731562a7ce.png">

step 3 しまった、ヒストリバックしちゃった！！
<img width="1035" alt="3" src="https://user-images.githubusercontent.com/1148320/215001803-32a2382c-cf56-455f-8933-fad0c6544b22.png">

step 4. 「→」ボタンで編集画面に戻ったけど、途中まで書いた記事が消えてる〜！！😱
<img width="1035" alt="4" src="https://user-images.githubusercontent.com/1148320/215001823-483d78ca-ecaa-4f0f-8bb3-bead81b839ad.png">

いちおう自動保存されているので、この状態で何もせず画面をリロードすれば途中まで書いてた内容は元に戻ります（今回はこれで助かった）。
ですが、あせって何か書き込んでしまうと、その内容で上書きされてしまうので、途中まで書いてた内容は失われてしまいます。

「えっ、消えた！？」と焦るのは精神衛生上よくないですし、自動保存されていた内容を予期せず上書きして消してしまうのはさらに悲惨なので、自動保存されていたとしてもヒストリバックしたり、タブを閉じたりしようとしたときはアラートを出してほしいです。（もしくはstep 3や4で、最新の内容が即座に反映されているとか）

## 参考情報
はてなブログ（有料版限定？）だと自動保存かつヒストリバックでアラート表示をやっているので、仕様の参考になるかもしれません。

**Comment by tanaka-asami415** [comment]
@JunichiIto 
ありがとうございます🙏
下書きが消えてしまった！というご意見はよく見かけており
我々も改善したいと思っておりますので、こちらも参考にさせていただきます！🙇

**Comment by wataru86** [comment]
@JunichiIto 

お待たせしました！
エディタページからのページ遷移時に警告を表示するように変更しました :rocket: 
また、別のページからエディタページに戻ってきた際に、最新の内容が反映されるように変更も行いました。

https://github.com/increments/qiita-discussions/issues/497

**Reply by JunichiIto** [reply]
ご対応＆ご連絡どうもありがとうございます！
